mewarpx/mewarpx/utils_store/init_restart_util.py

```diff
@@ -0,0 +1,91 @@
+    if os.path.isdir("diags"):
+        for d in next(os.walk("diags"))[1]:
+            if d.startswith(prefix):
+                print(f"Removing old checkpoint file {d}")
+                shutil.rmtree(os.path.join("diags", d))
+
+def run_restart(checkpoint_prefix="chkpoint", force=False, additional_steps=None):
+    '''
+    Attempts to restart a run by looking for checkpoint files
+    starting with a prefix in the diags directory.
+
+    Arguments:
+        checkpoint_prefix (string): Look for a checkpoint directory starting with this prefix
+          to restart from.
+        force (bool): If true, a problem with restarting from a checkpoint will cause an error,
+          otherwise simply print a warning.
+        additional_steps (int): The number of steps to run after restarting from the checkpoint.
+          If this is None then it will run to the current value of mwxrun.simulation.max_steps.
+    '''
+    # import must be done here to avoid a circular import
+    from mewarpx.mwxrun import mwxrun
+
+    logger.info(
+        "Attempting to " + ("force a " if force else "") +
+        "restart from the most recent checkpoint"
+        f"starting with '{checkpoint_prefix}'"
+    )
+
+    if not os.path.isdir("diags"):
+        if force:
+            raise RuntimeError("Diags directory does not exist!")
+        else:
+            logger.warning("Diags directory does not exist!")
+            return False
+
+    # using next() gives only the first layer of subdirectories
+    checkpoints = sorted(
+        [f for f in next(os.walk("diags"))[1]
+        if "old" not in f and f.startswith(checkpoint_prefix)]
+    )
+
+    if not checkpoints:
+        if force:
+            raise RuntimeError(
+                "There were no checkpoint directories "
+                f"starting with {checkpoint_prefix}!"
+            )
+        else:
+            logger.warning(
+                "There were no checkpoint directories "
+                f"starting with {checkpoint_prefix}!"
+            )
+            return False
+
+    checkpoint = checkpoints[-1:][0]
+    max_steps = mwxrun.simulation.max_steps
+    checkpoint_step = int(checkpoint.replace(checkpoint_prefix, ""))
+
+    if additional_steps is None:
+        mwxrun.simulation.max_steps = max_steps - checkpoint_step
+        print("MAX STEPS ", mwxrun.simulation.max_steps)
+        if mwxrun.simulation.max_steps == 0:
+            logger.warning(
+                f"The checkpoint directory was created at step {checkpoint_step}, "
+                f"but the max steps is also {max_steps}, so the simulation will "
+                f"only rerun step {checkpoint_step}."
+            )
+        if mwxrun.simulation.max_steps < 0:
+            raise RuntimeError(
+                f"The checkpoint directory was created at a later step ({checkpoint_step}) than the current max steps ({max_steps})!"
+            )
+    else:
+        mwxrun.simulation.max_steps = additional_steps
```

Comment:
Just to be sure, how does this work if the checkpoint happened at step x? Would it then run up to step = additional_steps or step = (x + additional_steps)?

Reply:
it runs to x + additional_steps. `max_steps` is the number of steps to run, not the step at which to stop.